apps/smart-contracts/token/test/RestrictedTransferHook.test.ts

```diff
@@ -244,4 +245,216 @@ describe('RestrictedTransferHook', () => {
       expect(await restrictedTransferHook.getBlocklist()).to.eq(JUNK_ADDRESS)
     })
   })
+  describe('hook', () => {
+    let source: SignerWithAddress
+    let destination: SignerWithAddress
+    beforeEach(async () => {
+      await setupHookAndLists()
+      await restrictedTransferHook.connect(owner).setToken(ppoToken.address)
+      source = user1
+      destination = user2
+    })
+
+    it('reverts if caller is not PPO', async () => {
+      expect(await restrictedTransferHook.getToken()).to.not.eq(user1.address)
+
+      await expect(
+        restrictedTransferHook.connect(user1).hook(source.address, destination.address, 1)
+      ).to.be.revertedWith('msg.sender != token')
+    })
+
+    describe('if source blocked', () => {
```

Comment:
```suggestion
    describe('if sender blocked', () => {
```
so we don't overlap in terminology

please also update the terminology in every test below so it specifically uses one of: sender, recipient, source, destination

Reply:
I do not understand why we need four names? A transfer will always be between two participants. 
Either we can call them `sender`, `recipient` or `source`, `destination`. 

The recipient/destination is going to be in `blocklist` (in this case, revert) or in `allowedDestinations`(in this case succeed) or both (in this case also, revert).

In order to avoid confusion of Recipient vs Destination, in the contract we can have:

```
    require(!_blockedAccounts.isIncluded(_from), "Sender blocked");
    require(!_blockedAccounts.isIncluded(_to), "Recipient blocked");
    if (_allowedSources.isIncluded(_from)) return;
    require(_allowedDestinations.isIncluded(_to), "Recipient not allowed");
```

or 

```
    require(!_blockedAccounts.isIncluded(_from), "Source blocked");
    require(!_blockedAccounts.isIncluded(_to), "Destination blocked");
    if (_allowedSources.isIncluded(_from)) return;
    require(_allowedDestinations.isIncluded(_to), "Destination not allowed");
```